public/js/redux/chartsRedux.js

```diff
@@ -2,23 +2,47 @@ import { State, Actions, Effect } from 'jumpstate';
 import api from 'services/Api';
 import chartList from 'utils/chartList';
 import { createYTotalsList, createPartialsList, formattedSeries, compareDates, fillMissingDates } from 'helpers/chartsHelpers';
+import { reEstablishSession } from 'panda-session';
 import moment from 'moment';
 
 /* ------------- State Management ------------- */
 
+const updateAttemptActions = (chartData, chart, startDate, endDate) => {
+    Actions.toggleIsUpdatingCharts(false);
+    Actions[`update${chart}`]({ chartData, startDate, endDate });
+};
+
+const responseFailActions = (chart, error) => {
+    Actions.toggleIsUpdatingCharts(false);
+    Actions[`get${chart}Failed`](error);
+};
+
 Effect('filterDesk', (filterObj) => {
     Actions.updateFilter(filterObj);
     Actions.toggleIsUpdatingCharts(true);
     const { startDate, endDate, desk, productionOffice } = filterObj;
     chartList.map(chart => {
         api[`get${chart}`](startDate, endDate, desk, productionOffice)
```

Comment:
This should be parameterised ideally as it gets called twice so it would be very easy to forget to change one of the lines. It wouldn't be obvious the second call hadn't been changed until someone got a 419 which makes it a hard bug to replicate.